runtime-decompiler/src/main/java/org/jrd/frontend/PluginMangerFrame/PluginConfigurationEditorController.java

```diff
@@ -63,6 +68,57 @@ public PluginConfigurationEditorController(PluginConfigurationEditorView view, P
         view.getOkCancelPanel().getCancelButton().addActionListener(actionEvent -> {
             view.dispose();
         });
+        view.getOkCancelPanel().getValidateButton().addActionListener(actionEvent -> {
+            if(view.getPluginListPanel().getWrapperJList().getSelectedIndex() == -1) return;
+            JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+            //preparing data for the compiler
+            List<URL> dependencyURLs = ((DecompilerWrapperInformation)view.getPluginListPanel().getWrapperJList().getSelectedValue()).getDependencyURLs();
+            ArrayList<String> compileStringA = new ArrayList<>();
+            compileStringA.add("-cp");
+            StringBuilder dependencyS = new StringBuilder();
+            dependencyS.append("\"");
+            for (URL dependency: dependencyURLs)
+            {
+                dependencyS.append(":").append(dependency.getPath());
+            }
+            dependencyS.append("\"");
+            compileStringA.add(dependencyS.toString());
+            compileStringA.add("-d");
+            compileStringA.add("/tmp");
+            compileStringA.add(((DecompilerWrapperInformation) view.getPluginListPanel().getWrapperJList().getSelectedValue()).getWrapperURL().getPath());
```

Comment:
this looks heavily like copypasted code. Please wxtract javac call to different class